services/Authentication/src/Authentication.BusinessLogicLayer/Services/Utility/UserContext.cs

```diff
@@ -0,0 +1,8 @@
+using Authentication.DataAccessLayer.Entities.Models;
+
+namespace Authentication.BusinessLogicLayer.Services.Utility;
+
+public class UserContext
+{
+    protected internal User? User;
+}
```

Comment:
why do you need this class?

Reply:
I needed to ensure that the TokenService and AuthenticationService access the same User object. The User object is needed to save the currently authenticated user. I settled on the solution of creating a class and injecting it into each of the services using DI. Are there more elegant solutions to this problem?

Reply:
you can use HttpContext.User

Reply:
You're right, that should be a better option, i'll try it